proposed/simplecache.md

```diff
@@ -212,24 +212,26 @@ interface CounterInterface
     /**
      * Increment a value atomically in the cache by the given step value and return the new value
      *
-     * If the key does not exist, it is initialized to the value of $step
+     * If the key does not exist, it is initialized to the value of $step.
+     * If the value is increased above PHP_INT_MAX the return value is undefined.
      *
      * @param string $key  The cache item key
      * @param int    $step The value to increment by, defaulting to 1
      *
-     * @return int|bool The new value on success and false on failure
+     * @return int|false The new value on success and false on failure
      */
     public function increment($key, $step = 1);
 
     /**
      * Decrement a value atomically in the cache by the given step value and return the new value
      *
-     * If the key does not exist, it is initialized to the value of -$step
+     * If the key does not exist, it is initialized to the value of -$step.
+     * If the value is decreased below PHP_INT_MIN the return value is undefined.
```

Comment:
The problem will be to detect the cases since not all cache implementations deal with this the same way. Checking it on each call will create overhead without much gain. Therefor I'm in fav. that these edge cases should be handled at application level if needed.